api/v1/overlaynetwork_types.go

```diff
@@ -20,34 +20,47 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
 // OverlayNetworkSpec defines the desired state of OverlayNetwork
 type OverlayNetworkSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Name is the NetworkAttachmentDefinition name.
+	Name string `json:"name"`
```

Comment:
Please explain why this field is not namespaced. The documentation you linked in the PR description has the explanation.

Reply:
I am not sure how that document explains it.

We actually can drop this parameter and use the CR name (and namespace) to create the NAD.
 We can also rename `NetworkName` to `Name`.

Reply:
Done